### ISSUE TITLE
EAP 225 (v3)

### ISSUE THREAD

**Issue by Loki321** [ericpignet/home-assistant-tplink_router#14]
This doesn't seem to be working on my EAP-225 v3 AP (on latest firmware);

`Log Details (ERROR)
Logger: custom_components.tplink_router.device_tracker
Source: custom_components/tplink_router/device_tracker.py:473
Integration: tplink_router (documentation)
First occurred: 9:15:36 PM (1 occurrences)
Last logged: 9:15:36 PM

AP didn't respond with JSON. Check if credentials are correct`

The credentials are correct. I have tried with the current version and the previous.

**Comment by ericpignet** [comment]
Was it working with previous Home Assistant versions, or has it never worked?

**Comment by Loki321** [comment]
It has never worked, I moved over from a different router a couple of months ago and tried it then and have never been able to get it to work. Does the v3 have a different interface maybe? Happy to give you temporary access if it would help.

**Comment by Loki321** [comment]
I just noticed you asked about home-assistant versions (I assumed you meant tp-link component versions), I am running 0.114.4 and it's the only version I've tried this component with, I'm avoiding upgrading at the moment because I don't really have the time to deal with the breaking changes, is the component meant to work with version 3 hardware? If so, I'll try the upgrade.

I can give you secure access to the EAP web interface through a reverse proxy if it is different.

**Comment by nervx** [comment]
Same problem for me :/.

**Comment by nervx** [comment]
Problem solved i think.

Url has changed between EAP225 V2 & EAP225 V3 and some json parameters too :

Line 459 i changed "/data/monitor.client.client.json" by "/data/status.client.user.json"
Line 479 i changed "device['MAC'].replace('-', ':'): device['DeviceName']" by "device['MAC'].replace('-', ':'): device['hostname']"

And now in HA, i have some device like it :

device_tracker.31_A1_7b_eb_cc_55
source_type: router
latitude: XX.X
longitude: XX.X
gps_accuracy: 0
scanner: EAP225TplinkDeviceScanner
friendly_name: 31_A1_7b_eb_cc_55

**Comment by Loki321** [comment]
Excellent work, I've just made the same changes and it is working. The only problem is it's now not possible to stay logged in through the web interface but it's rare I need to make changes so I can work around that.

Thanks.

**Comment by nervx** [comment]
Yes i don't know if it's possible to change that in settings but on the main page of this github repository, it's written :

> TP-Link devices typically only allow one login at a time to the admin console. This integration will count towards your one allowed login. Depending on how aggressively you configure device_tracker you may not be able to access the admin console of your TP-Link device without first stopping Home Assistant. Home Assistant takes a few seconds to login, collect data, and log out. If you log into the admin console manually, remember to log out so that Home Assistant can log in again. 

I will wait but currently, it seems that my devices don't update their states and you ?

**Comment by Loki321** [comment]
I've just tested by putting my phone on airplane mode and it changes to 'not_home', changing it back goes back to 'home' so it seems to be working correctly so far.

Edit: It takes a couple of minutes to go to 'not_home' but that has been the case with all of the router based device trackers I've used, changing to 'home' is pretty quick.

**Comment by nervx** [comment]
Yes, you have reason. I thought the attribute "last_updated" changed for each device when the the device tracker system do a request to the tp link router.

So it's perfect, i can now add a new condition to my automation and to be sure there is no people at home and launch my vacuum :)
